torch_xla/test/metrics_test_wrapper.py

```diff
@@ -0,0 +1,131 @@
+"""Wrap PyTorch XLA training jobs with extra logic around XLA metrics.
+
+It can be used to save a historical record of metrics and/or check metrics
+against a golden set of metrics to search for performance regressions.
+
+This script expects a certain directory structure to setup tests and to hold
+metrics results. In the outermost directory, there should be a base config file
+detailed below. Within that directory, there should be a subdirectory for each
+test that contains a metrics_history subdirectory and (optionally) a config file
+to override fields of the base config. In other words, it should look like this:
+
+    root/
+      base_config.json
+      mnist/
+        config_overrides.json
+        metrics_history/
+
+TODO(zcain): Design and document the config file.
+
+Example usage:
+python metrics_test_wrapper.py --root="gs://model_metrics" \
+    --test_folder_name="mnist" -- python test/test_train_mnist.py --num_epochs=1
+"""
+import argparse
+import datetime
+import glob
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import torch_xla.utils.gcsfs as gcsfs
+
+
+_CLOUD_STORAGE_PREFIX = 'gs://'
+_METRICS_HISTORY_DIR_NAME = 'metrics_history'
+_XLA_METRICS_FILE = 'XLA_METRICS_FILE'
+
+
+def _find_correct_metrics_file(base_metrics_file):
+  # Distributed training jobs write 1 file per core. If XLA_METRICS_FILE is set
+  # to 'base_metrics_file', then the distributed output files have the format
+  # 'base_metrics_file.0', 'base_metrics_file.1', etc.
+  metrics_files = glob.glob('{}*'.format(base_metrics_file))
+  if len(metrics_files) == 1:
+      # Non-distributed case: the correct file is simply 'base_metrics_file'.
+    file_to_read = base_metrics_file
+  else:
+    # Choose the file of the first core. When sorted alphabetically, the first
+    # element will be 'base_metrics_file' which is empty in the distributed
+    # case. The second element will be 'base_metrics_file.0'.
+    file_to_read = sorted(metrics_files)[1]
```

Comment:
I think we should make this logic a bit more robust.
Like checking files size or something along those lines.

Reply:
Sounds good, switched to that